nodesharing-lib/src/main/java/com/redhat/jenkins/nodesharing/ConfigRepo.java

```diff
@@ -92,16 +92,30 @@ public ConfigRepo(@Nonnull String url, @Nonnull File workingDir) {
         Files.createDirectories(workingDir.toPath());
         TaskLog taskLog = new TaskLog(new File(workingDir.getAbsolutePath() + ".log"));
         try {
-            ObjectId currentHead = getRemoteHead(taskLog);
             synchronized (repoLock) {
-                if (snapshot != null && currentHead.equals(snapshot.source)) {
-                    LOGGER.fine("No config update in " + url + " after: " + snapshot.source.name());
-                } else {
-                    taskLog.getLogger().printf("Node sharing config changes discovered %s%nPulling %s to %s%n", currentHead.name(), url, workingDir);
-                    fetchChanges(taskLog);
-                    ObjectId checkedOutHead = getClient(taskLog).revParse("HEAD");
-                    assert currentHead.equals(checkedOutHead): "What was discovered was in fact checked out";
-                    snapshot = readConfig(currentHead, taskLog);
+                ObjectId currentHead = null;
+                try {
+                    currentHead = getRemoteHead(taskLog);
+                } catch (GitException e) {
+                    LOGGER.info("Getting HEAD of config repo from remote location failed");
+                    if (snapshot != null) {
+                        LOGGER.info("Trying to recover from previous locally stored config");
+                        snapshot = readConfig(snapshot.source, taskLog);
```

Comment:
Since `readConfig` can throw, we should better attach the `GitException` as suppressed here.